apps/coscrad-frontend/src/components/tags/tag-card.tsx

```diff
@@ -0,0 +1,60 @@
+import { ITagViewModel } from '@coscrad/api-interfaces';
+import { LinkSharp as LinkIcon } from '@mui/icons-material';
+import {
+    Card,
+    CardActionArea,
+    CardActions,
+    CardContent,
+    CardHeader,
+    Divider,
+    IconButton,
+} from '@mui/material';
+import { Link } from 'react-router-dom';
+import { CategorizableOfMultipleTypeContainer } from '../higher-order-components';
+import { thumbnailCategorizableDetailPresenterFactory } from '../resources/factories/thumbnail-resource-detail-presenter-factory';
+
+export const TagCard = ({ id, label, members }: ITagViewModel): JSX.Element => {
+    const allResourceTypesWithRepeats = members.flatMap(({ type }) => type);
+
+    const uniqueResourceTypesWithThisTag = [...new Set(allResourceTypesWithRepeats)];
+
+    const initialEmptyReosurceTypesAndSelectedIds = uniqueResourceTypesWithThisTag.reduce(
+        (acc, resourceType) => ({
+            ...acc,
+            [resourceType]: [],
+        }),
+        {}
+    );
+
+    // A Map might improve readability here
+    const resourceTypesAndSelectedIds = members.reduce(
```

Comment:
Can we make the CategorizableOfMultipleTypesContainer responsible for this parsing? We want to do it once and make things easy for the client of the util.

Reply:
Let's do this after all open branches are merged down to avoid conflicts.

Reply:
I've added a todo in the code.